autoextract_poet/pages.py

```diff
@@ -0,0 +1,38 @@
+from dataclasses import dataclass
+
+from web_poet.pages import ItemPage
+
+from autoextract_poet.mixins import AutoExtractResponseShortcutsMixin
+from autoextract_poet.inputs import (
+    AutoExtractProductResponseData,
+    AutoExtractProductListResponseData,
+)
+
+
+@dataclass
+class AutoExtractProductPage(ItemPage, AutoExtractResponseShortcutsMixin):
+    """Generic product page."""
+
+    autoextract_response: AutoExtractProductResponseData
+
+    def to_item(self):
+        product = self.autoextract_response.data.get("product", {})
+        return product
+
+
+@dataclass
+class AEProductListPage(ItemPage, AutoExtractResponseShortcutsMixin):
+    """Generic product list page."""
+
+    autoextract_response: AutoExtractProductListResponseData
+
+    def breadcrumbs(self):
+        return self.autoextract_response.data.get("productList", {}).get("breadcrumbs", [])
```

Comment:
Just some rant :joy: 
Can we do better than `obj.get('some key', default).get('other key', default)` and have it as 
`obj['some key']['other key']`, basically having the defaults under the hood?
Not a problem here, just thinking about it..

Reply:
Maybe, having `jmes` like we have `xpath` and `css` for regular HTML responses

Reply:
I've refactored this class to make use of a private property. I was not including another dependency (jmes) for this simple use case xD But thank you for the suggestion.